src/api.js

```diff
@@ -64,6 +64,37 @@ export async function getUserFollowedArtists(token) {
   return artists
 }
 
+/**
+ * Return the artists whose songs the user has liked
+ *
+ * @param {string} token
+ * @returns {Promise<Artist[]>}
+ */
+export async function getUserLikedSongArtists(token) {
+  /** @type {Artist[]} */
+  const artists = []
+  const params = new URLSearchParams({ limit: String(50) })
+
+  let next = apiUrl(`me/tracks?${params}`)
+
+  while (next) {
+    /** @type Paged<SpotifySavedTrack> */
+    const response = await get(next, token)
+
+    // This is a 2D array of tracks with nested artists
+    const artistsByTrack = response.items.map((item) => item.track.artists).flat()
+    const nextArtists = artistsByTrack.map(buildArtist)
+
+    artists.push(...nextArtists)
+    next = response.next
+  }
+
+  // Remove duplicate artists
+  const uniqueArtists = [...new Set(artists)]
```

Comment:
Instead of converting array => set => array, you can make this more efficient by skipping the first conversion: `const artists = new Set()` and then `for (const artist of nextArtists) artists.add(artist)`

Reply:
Actually, I think using `Set` won't work here, because `buildArtist` always create a new object. Therefore, the reference check (used by `Set` internally) is always false, even for the same artist.